packages/_vue3-migration-test/src/components/test-layouts-composable.vue

```diff
@@ -0,0 +1,39 @@
+<template>
+  <header v-if="hasContent('header')">
+    <!-- @slot Slot that is used to insert content into the Header. -->
+    <slot name="header">
+      <span v-if="devMode">HEADER</span>
+    </slot>
+  </header>
+
+  <div v-if="hasContent('sub-header')">
+    <!-- @slot Slot that can be used to insert content into the Sub Header. -->
+    <slot name="sub-header">
+      <span v-if="devMode">SUB HEADER</span>
+    </slot>
+  </div>
+
+  <main v-if="hasContent('main')" class="x-layout__main">
+    <!-- @slot Slot that can be used to insert content into the Main. -->
+    <slot name="main">
+      <span v-if="devMode">MAIN</span>
+    </slot>
+  </main>
+</template>
+
+<script setup>
+  import { useLayouts } from '../../../x-components/src/components/layouts/use-layouts';
+
+  const slots = defineSlots();
+
+  const props = defineProps({
+    devMode: {
+      type: Boolean,
+      default: true
+    }
+  });
+
+  const { hasContent } = useLayouts(props.devMode, slots);
```

Comment:
It is not working as expected. `useLayouts` needs the `context` as second parameter to do, then `context.slots['header']`. So in this example will be `undefined` as we are passing here only the `slots`. But as `devMode` is enabled, the rest of the composable is not evaluated. 

Try to make `devMode` as `false` and you can see better the issue:
![Screenshot 2024-05-27 at 11 49 49 AM](https://github.com/empathyco/x/assets/10746604/ded396af-c34b-4c77-b6be-f360ccd16501)